File: 1-Easy/Horse-racing Duals/horse-racing-duals.c
#include <stdio.h>
#include <stdlib.h>

void fillHorseStrengths(int *horseStrengths, int horseCount);
int sortByAscending(const void *next, const void *current);
int getClosestDifference(int *horseStrengths, int horseCount);

int main()
{
    int horseCount;
    scanf("%d", &horseCount);
    int horseStrengths[horseCount];
    fillHorseStrengths(horseStrengths, horseCount);
    
    // Sorts the horseStrengths array in ascending order
    qsort(horseStrengths,
        sizeof(horseStrengths) / sizeof(*horseStrengths),
        sizeof(*horseStrengths),
        sortByAscending
    );
    
    int min = getClosestDifference(horseStrengths, horseCount);
    printf("%d", min);

    return 0;
}

// Populates the horseStrengths array
void fillHorseStrengths(int *horseStrengths, int horseCount)
{
    for (int i = 0; i < horseCount; i++)
    {
        scanf("%d", &horseStrengths[i]);
    }
}

// Comparator function for ascending order sorting
int sortByAscending(const void *x, const void *y)
{
    return *((int*)x) - *((int*)y);
}

// Iterates over the horseStrengths array, substracting the next element from the current one and returns the closest difference
int getClosestDifference(int *horseStrengths, int horseCount)
{
    int min = __INT_MAX__;

    for (int i = 0; i < horseCount - 1; i++)
    {
        int current = horseStrengths[i + 1] - horseStrengths[i];
        
        if (current < min)
        {
            min = current;
        }
    }

    return min;
}

File: 1-Easy/Horse-racing Duals/horse-racing-duals.java
import java.util.Scanner;
import java.util.Arrays;

class Solution {

    public static void main(String args[]) {
        Scanner scanner = new Scanner(System.in);
        int horseCount = scanner.nextInt();
        int[] horseStrengths = new int[horseCount];
        fillHorseStrengths(horseStrengths, horseCount, scanner);
        Arrays.sort(horseStrengths); // Sorts the horseStrengths array in ascending order
        
        int min = getClosestDifference(horseStrengths, horseCount);
        System.out.println(min);
    }

    // Populates the horseStrengths array
    static void fillHorseStrengths(int[] horseStrengths, int horseCount, Scanner scanner) {
        for (int i = 0; i < horseCount; i++)
        {
            horseStrengths[i] = scanner.nextInt();
        }
    }

    // Iterates over the horseStrengths array, substracting the next element from the current one and returns the closest difference
    static int getClosestDifference(int[] horseStrengths, int horseCount) {
        int min = Integer.MAX_VALUE;

        for (int i = 0; i < horseCount - 1; i++)
        {
            min = Math.min(min, horseStrengths[i + 1] - horseStrengths[i]);
        }

        return min;
    }
}